core/domain/exp_services.py

```diff
@@ -1710,6 +1710,475 @@ def is_voiceover_change_list(change_list):
+                                                     'widget_customization_args'
+                                                    ]) and
+                                    change.property_name not in
+                                    changed_translations[old_state_name]):
+                                change_is_mergeable = True
+                    else:
+                        change_is_mergeable = True
+                elif (change.property_name ==
+                      exp_domain.STATE_PROPERTY_INTERACTION_DEFAULT_OUTCOME
+                     ):
+                    if old_state_name in changed_properties:
+                        if (change.property_name not in
+                                changed_properties[old_state_name] and
+                                change.property_name not in
+                                changed_translations[old_state_name]):
+                            change_is_mergeable = True
+                    else:
+                        change_is_mergeable = True
+                elif (change.property_name ==
+                      exp_domain.STATE_PROPERTY_UNCLASSIFIED_ANSWERS):
+                    change_is_mergeable = True
+                elif (change.property_name ==
+                      exp_domain.STATE_PROPERTY_NEXT_CONTENT_ID_INDEX):
+                    change_is_mergeable = True
+                elif (change.property_name ==
+                      exp_domain.STATE_PROPERTY_LINKED_SKILL_ID):
+                    change_is_mergeable = True
+                # We’ll not be able to handle the merge if changelists
+                # affect the different indices of the hint in the same
+                # state because whenever there is even a small change
+                # in one field of any hint, they treat the whole hints
+                # list as a new value.
+                # So it will not be possible to find out the exact change.
+                elif (change.property_name ==
+                      exp_domain.STATE_PROPERTY_INTERACTION_HINTS):
+                    if old_state_name in changed_properties:
+                        if (change.property_name not in
+                                changed_properties[old_state_name] and
+                                change.property_name not in
+                                changed_translations[old_state_name]):
+                            change_is_mergeable = True
+                    else:
+                        change_is_mergeable = True
+                elif (change.property_name ==
+                      exp_domain.STATE_PROPERTY_INTERACTION_SOLUTION):
+                    if old_state_name in changed_properties:
+                        if (frontend_version_exploration.states[old_state_name].interaction.id == # pylint: disable=line-too-long
+                                backend_version_exploration.states[new_state_name].interaction.id): # pylint: disable=line-too-long
```

Comment:
Here and elsewhere it seems like `*_version_exploration.states[..]` makes the lines too long. Can we use a new variable for the states? e.g. frontend_exp_states = frontend_version_exploration.states[..] and backend_exp_states = ..?

Reply:
Yep. Made the changes.